addons/stock/models/stock_inventory.py

```diff
@@ -452,7 +452,8 @@ def create(self, vals_list):
         set in an onchange of `product_id`.
         Finally, this override checks we don't try to create a duplicated line.
         """
-        for values in vals_list:
+        products = self.env['product.product'].browse(map(lambda vals: vals['product_id'], vals_list))
```

Comment:
```suggestion
        products = self.env['product.product'].browse([vals.get('product_id') for vals in vals_list])
```
is slightly faster according to https://stackoverflow.com/questions/1247486/list-comprehension-vs-map/1247490#1247490. I also suggest to use `get()` as we did `if 'product_id' in value` below

Reply:
@Whenrow Yeah ok it's true that list comps are usually slightly faster than map.